### ISSUE TITLE
Add Tag to Branch with composer.json?

### ISSUE THREAD

**Issue by astorm** [adldap/adLDAP#69]
Would it be possible to tag a version of the repository that contains `composer.json`?  (4.05?)

The v4.04 tagged branch

https://github.com/adldap/adLDAP/tree/v4.0.4

doesn't have a `composer.json` file.  

This means I can't use composer to add this repository to my projects without either

1. Accepting a minimum stability that would let me target `dev-master`/the master branch
2. Forking the repository myself and using that project

Also -- with a tagged stable version of the official project out there, it might help with the current packagist  weirdness. 

Happy to provide more context if that would help.  Thank you for your time/attention.

**Comment by strebl** [comment]
I created a tagged version a while ago: https://packagist.org/packages/strebl/adldap

**Comment by samwilson** [comment]
Have a read through #4 — the chances seem to small that anything's going to be done to fix the situation.

@stevebauman is also maintaining a fork; see https://github.com/adldap/adLDAP/issues/4#issuecomment-71497756 In the interests of slightly reducing the already-prevalent fragmentation, I'd recommend using that one.

**Comment by samwilson** [comment]
(Oh, and I'd say this issue could be closed as a dupe of #4.)

**Comment by stevebauman** [comment]
Yea to be honest, this repository is definitely abandoned in my opinion. The main packagist author has been inactive from github for close to a year and rich2k has been inactive for well over a year.

I'd suggest using a fork that's available on packagist and verify the authors recent commits so you know it'll be in the hands of someone who is active in the community. 

Personally, I'm going to start combing through this giant mess of forks and start merging in valid fixes, and hopefully create a better adLDAP from it.

**Comment by strebl** [comment]
@stevebauman :+1: 
As soon as you are ready i'll abandon my package and replace it with yours!
I just didn't see your package. Maybe because the name is different.

I was going to add some functionality to my package. But if you'll maintain your repo, I'll create the PRs on your repo!?

**Comment by stevebauman** [comment]
@strebl That's no problem, and I'll be starting the merges shortly, I'm just cleaning up the classes right now and I'll be all set. Feel free to create issues and PRs on my forked repo!

**Comment by astorm** [comment]
@stevebauman At the risk of volunteering someone for extra work -- it'd be great if your repository had the same tag history as https://github.com/ztec/adLDAP.git

With that tag history in place there'd be a more compelling case for taking over the `adldap/adldap` packagist name i.e. preserver the current versions, but give current  `adldap/adldap`  users the chance to upgrade.

**Comment by stevebauman** [comment]
@astorm I agree with you, that's a great idea. I'll create a separate branch with the old history and tag the previous releases.

I use this library myself like many of you, and want to continue using and improving it, so please don't hesitate making suggestions and changes that need to be made for the better of the library.

**Comment by laurin1** [comment]
We're using @stevebauman 's version, and I'm all for moving in that direction as well and would be interesting in contributing as well - adLDAP is a critical piece of our operation.

**Comment by strebl** [comment]
Yeah, the adldap/adldap name would be fantastic.

**Comment by stevebauman** [comment]
Unfortunately getting the original packagist repository name is a slim to none chance as @ztec has already said he would do this and didn't deliver. Unless there's some way of 'petitioning' the name then I'm not sure if we can recover it as far as my knowledge goes.

Though I also wanted to say, I'll be able to add other maintainers to this fork as well, hopefully further ensuring adLDAPs future.

**Comment by strebl** [comment]
@stevebauman Yah, you are right. Maybe it's only me, but I don't like the -fork suffix. It takes some value off of your package.

I'll try my best to help you as good as I can.

**Comment by astorm** [comment]
@stevebauman Petitioning for the name is sort of what I had in mind -- being able to come to the packagist folks with a backwards compatible repository **with** recent commits and some interest might fly and/or might get the attention of the folks currently ignoring the other repositories (leading to them merging in fixes)

**Comment by ralfbecker** [comment]
I'm also maintaining an adLDAP fork for http://www.egroupware.org/ and have (unsuccessful) sent pull requests with important fixes to adLDAP. I'm willing to help maintaining a forked adLDAP, as I have to maintain it anyway. Name-wise I would probably go for adLDAP2, as it makes it immediately clear what to use ;-)

**Comment by stevebauman** [comment]
@strebl You're right, I don't like the -fork suffix either, and I agree with your reasoning. 

@ralfbecker I agree with adLDAP2 as well! I'll go with this.

@astorm If we use the name adLDAP2 as ralf has suggested, then we won't need to recover the previous name. We can keep namespaces the same and work on the classes itself and upgrade the code base as necessary, allowing users to migrate to adLDAP2 without issues.

I've created a new adLDAP2 organization. I'm going to migrate everything over to this repository. Give me a few moments to get everything set up. https://github.com/adLDAP2/adLDAP2

**Comment by samwilson** [comment]
Thanks @stevebauman! This is terrific.

**Comment by astorm** [comment]
@stevebauman Again, I'm always hesitant to volunteer extra work -- but a tag compatible version means we **might** be able to convince the packagist folks to hand over `adldap/adldap` to the new repository -- which would be a win for the 14,000ish folks who've installed the library and (more importantly) help prevent a lot of confusion from folks who are just installing the first thing they see.

**Comment by stevebauman** [comment]
@astorm Has this been done before? I've never experienced this sort of issue. Do I need to contact packagist personally once I've completed making tag compatible versions?

**Comment by strebl** [comment]
Does anyone know if there is a way to compare my adldap fork to the adldap2 repo using github or another solution?

**Comment by ChadSikorra** [comment]
Has anyone tried to actually email Rich or Scott to see if they really have abandoned their work on this library? It looks like they do have their email addresses on the front page and it appears like they did mean to release a 5.0.0 version at some point.

**Comment by stevebauman** [comment]
Both Rich and Scott have both left the [adLDAP](https://github.com/adldap) organization they created, so presumably no, I don't think they're interested in working on it anymore.

**Comment by ChadSikorra** [comment]
Good catch. I didn't even notice that.

**Comment by wunc** [comment]
@stevebauman Dude, thanks so much for this! I had forked adLDAP just to get some customizations that I needed for our org: authenticating with something other than samaccountname or guid, better search options. Glad to see someone willing to reinvigorate the project.

**Comment by stevebauman** [comment]
No problem! If you're able to help definitely send some pulls my way. The more community and help, the better.
